lib/vagrant-parallels/guest_cap/windows/install_parallels_tools.rb

```diff
@@ -4,16 +4,26 @@ module GuestWindowsCap
       class InstallParallelsTools
         def self.install_parallels_tools(machine)
           machine.communicate.tap do |comm|
+            # Get the host arch. This is safe even if an older x86-only Vagrant version is used.
+            arch = `arch -64 uname -m`.chomp
+
             pti_agent_path = File.expand_path(
-              machine.provider.driver.read_guest_tools_iso_path('windows'),
+              machine.provider.driver.read_guest_tools_iso_path('windows', arch),
               machine.env.root_path
             )
 
-            remote_file = '$env:TEMP\PTIAgent.exe'
+            # Older versions of Parallels provided an executable file instead of an ISO image.
+            agent_path_is_exe = pti_agent_path.end_with?('.exe')
+
+            remote_file = "C:\\Windows\\Temp\\parallels-tools-win.#{agent_path_is_exe ? 'exe' : 'iso'}"
```

Comment:
Removed `$env:TEMP` as I couldn't get it to work. It couldn't find the uploaded file, not sure where it was putting it.